### PR TITLE
arpack-ng svd

### PR DESCRIPTION
**Pull request by jurajHasik**
svd solver using arpack-ng svd for double-precision real matrices

### CODE REVIEW

File: include/pi-peps/linalg/arpack-rcdn.h [hunk 1]
```diff
@@ -183,4 +225,531 @@ struct ARDNS {
...
+               MatRef<Real> const& U,
+               VectorRef const& D,
+               MatRef<Real> const& V,
+               Args const& args) {
+      bool dbg = args.getBool("svd_dbg", false);
+
+      auto Mr = nrows(M);
+      auto Mc = ncols(M);
+      if (Mr < Mc) {
+        if (dbg)
```
Comment (asartori86) [inline include/pi-peps/linalg/arpack-rcdn.h:241]: should we use the macro `# ifdef DEBUG` also here?

File: include/pi-peps/linalg/arpack-rcdn.h [hunk 2]
```diff
@@ -183,4 +225,531 @@ struct ARDNS {
...
+
+        // std::cout<<"[avt]";
+        // for (int i=0; i<Mr; i++) std::cout<<" "<< x[i];
+        // std::cout<<std::endl;
+        // std::cout<<"[avt]";
+        // for (int i=0; i<Mc; i++) std::cout<<" "<< y[i];
+        // std::cout<<std::endl;
+      };
+
+      for (;;) {
```
Comment (asartori86) [inline include/pi-peps/linalg/arpack-rcdn.h]: maybe a `do while` is more readable

**Review by asartori86 — CHANGES_REQUESTED** [review]
ok great, just on more comment:
in the `meson.build` inside the `unit_test` folder, add the test that uses arpack only if the build flag has been set

**Comment by jurajHasik** [comment]
If would leave debugging to be selected at runtime 
through json input - by default turned off. The conditional compilation
of arpack test is in

**Review by asartori86 — APPROVED** [review]

**Pull request merged**